### DISCUSSION TITLE
adding gtf file

### DISCUSSION THREAD

**Discussion by dence** [GMOD/jbrowse-components#1545] [Q&A] [answered]
Hi, I'm trying to add a sorted and tabix-indexed gtf file as a track to a local jbrowse instance. I tried various values for the "type" option, but I'm not certain if any of them are recognized by jbrowse (see below). It would be helpful if in the add-track usage or in the config guide a list of valid values for the "type" option. The GTF file that I'm trying to add is available at https://treegenesdb.org/FTP/Genomes/Pita/v2.01/annotation/

Regards,
Daniel Ence

> (grep ^"#" Pita.2_01.gtf; grep -v ^"#" Pita.2_01.gtf | sort -k1,1 -k4,4n) | bgzip > Pita.2_01.sorted.gtf.gz
> tabix -p gff Pita.2_01.sorted.gtf.gz 
> jbrowse add-track Pita.2_01.sorted.gtf.gz --type=gtf --load copy
Error: Track type is not recognized
> jbrowse add-track Pita.2_01.sorted.gtf.gz --type=gff3 --load copy
Error: Track type is not recognized
> jbrowse add-track Pita.2_01.sorted.gtf.gz --type=gff --load copy
Error: Track type is not recognized
> jbrowse add-track Pita.2_01.sorted.gtf.gz --load copy
Error: Track type is not recognized

**Comment by cmdcolin** [comment, marked as answer]
Thanks for reporting

We don't currently support GTF/GTFTabix but this would be interesting to add.

It is also helpful to see the "back and forth dialog" that you have with the CLI as it makes it clear where our sort of weakness is

In this case, --type is actually trackType. This could be a candidate for renaming!

In the short term you could convert the GTF to GFF3 and then convert that to GFF3 tabix probably https://jbrowse.org/docs/faq.html#how-do-i-convert-gtf-to-gff

**Comment by cmdcolin** [comment]
(and by trackType I mean like manually specifying --type AlignmentsTrack for something like this. normally you would not have to specify this option since it is sort of auto-inferred from the file extension)

**Comment by dence** [comment]
Thank you Colin.
